synfig-studio/src/gui/docks/dockbook.cpp

```diff
@@ -260,6 +260,13 @@ DockBook::tab_button_pressed(GdkEventButton* event, Dockable* dockable)
 {
 	SYNFIG_EXCEPTION_GUARD_BEGIN()
 	CanvasView *canvas_view = dynamic_cast<CanvasView*>(dockable);
+
+	// Handle middle mouse click event first before showing the canvas
+	if (event->button == 2) {
```

Comment:
You need to check if `canvas_view` is available after `dynamic_cast`.

```
if (event->button == 2 && canvas_view) {
```

Reply:
There were two checks done (for middle click and left click) so I thought checking it right after casting would be appropriate https://github.com/synfig/synfig/pull/2107/commits/a2c90d3e36bb370c905e12b436cb9bc6f226db7f